uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerHost.java

```diff
@@ -201,8 +201,8 @@ private void performDeviceRegistrationHelper(@NonNull final String username) {
     private void postJoinConfirmHelper(@NonNull final String expectedUpn) throws InterruptedException {
         Logger.i(TAG, "Confirming that Shared Device Registration is successfull or not..");
 
-        // Adding brief wait to increase reliability of test
-        Thread.sleep(TimeUnit.SECONDS.toMillis(5));
+        // Adding wait to increase reliability of test
+        Thread.sleep(TimeUnit.SECONDS.toMillis(15));
```

Comment:
Quite a big wait again. I think this is fine for now if it helps make the tests reliable, but we should revisit these techniques in the future and explore alternatives that may offer better performance without compromising on reliability.

Reply:
I'm not sure how else you could handle something like this, unless we stopped relying on ui to confirm these things and perhaps found a way to do it programatically.

Reply:
We could think about using waitForExists() when applicable, that may reduce the need for some of this.

Reply:
Changed to waitForExists()